common/libs/framing/serial_listeners.h

```diff
@@ -0,0 +1,32 @@
+#ifndef __SERIAL_LISTENERS_H
+#define __SERIAL_LISTENERS_H
+
+enum RxError_t {
+  RX_ERROR_UNKNOWN,
+  RX_ERROR_OVR,
+  RX_ERROR_SERIAL_FRAMING,
+  RX_ERROR_TIMEOUT,
+  RX_ERROR_DMA
+};
+
+// An interface that gets called on serial events.
+// NOTE: all callbacks are called from interrupt context!
+class RxListener {
+public:
+  // Called on RX complete
+  virtual void onRxComplete() = 0;
```

Comment:
> Please make these no-ops rather than pure virtual. Adding pure virtual functions pulls in the whole C++ exceptions ABI which pulls in malloc.

What if we just compile with `-fno-exceptions`?  Pure virtual is nice because the compiler enforces that we implement every function.

Reply:
Unfortunately seems like it requires more effort than just `-fno-exceptions` - http://elegantinvention.com/blog/information/smaller-binary-size-with-c-on-baremetal-g/ .

Reply:
#413